core/src/main/java/org/bitcoinj/core/PeerAddress.java

```diff
@@ -282,6 +282,11 @@ protected void parse() throws ProtocolException {
                         addr = null;
                         break;
                     case I2P:
+                        if (addrLen != 32)
+                            throw new ProtocolException("invalid length of I2P address: " + addrLen);
+                        hostname = BASE32.encode(addrBytes).replace("=","") + ".b32.i2p";
```

Comment:
Why not use `omitPadding()` instead of the `.replace()`?

Reply:
> Why not use `omitPadding()` instead of the `.replace()`?

Because I didn't know it existed 🤦‍♂️

Also, I didn't know if any changes that affected the other address types should be made.

Reply:
If you use the `.omitPadding()` only on "your" code of block, it will stay local to that. Of course, #2642 suggests to use it in general in PeerAddress – also for `.onion`.